.github/workflows/solana-v2-rust.yml

```diff
@@ -0,0 +1,229 @@
+            ~/.cache
+          key: solana-${{ matrix.solana }}
+
+      - name: Install Solana
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/${{ matrix.solana }}/install)"
+          echo "$HOME/.local/share/solana/install/active_release/bin" >> $GITHUB_PATH
+
+      - name: Use Node ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Cache node dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ./solana/program_v2/node_modules
+          key: yarn-client-${{ hashFiles('solana/program_v2/yarn.lock') }}
+
+      - name: Install
+        run: yarn install --ignore-scripts --frozen-lockfile
+
+      - name: Lint
+        run: yarn lint
+
+      - name: Build program and run integration tests
+        run: |
+          yarn build
+#          yarn anchor build
+          yarn anchor test
+
+      # - name: Upload a single artifact
+      #   uses: actions/upload-artifact@v3
+      #   with:
+      #     path: /solana/program_v2
+      #     name: latest-program_v2_version
+      
+
+  E2E:
+    name: Test the gateway_v2 cli against the on-chain program on Solana ${{ matrix.solana }}, Node ${{ matrix.node }}, Rust ${{ matrix.rust }}, and ${{ matrix.os }}
+    needs: anchor-e2e-test
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        node: ["16.x"]
+        os: [ubuntu-latest]
+        solana: ["v1.10.29"]
+        rust: ["1.61"]
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+
+      - name: Use Rust ${{ matrix.rust }}
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+          override: true
+          profile: minimal
+
+      - name: Cache build dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            ./solana/program_v2/target
+          key: cargo-build-${{ hashFiles('program_v2/Cargo.lock') }}
+
+      - name: Cache Solana version
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache
+          key: solana-${{ matrix.solana }}
+
+      - name: Install Solana
+        run: |
+          sh -c "$(curl -sSfL https://release.solana.com/${{ matrix.solana }}/install)"
+          echo "$HOME/.local/share/solana/install/active_release/bin" >> $GITHUB_PATH
+      - name: Use Node ${{ matrix.node }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+
+      - name: Cache node dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ./solana/program_v2_cli/node_modules
+          key: yarn-cli-${{ hashFiles('program_v2_cli/yarn.lock') }}
+
+      - name: Install cli dependencies
+        run: yarn install
+        working-directory: solana/program_v2_cli
+
+      # - name: Download a single artifact
```

Comment:
Should we create a ticket for this ? (CC @rado0x54 )

Reply:
I think it would likely be worthwhile to have it...